lib/pavilion/builder.py

```diff
@@ -209,6 +212,15 @@ def __init__(self, pav_cfg, test, mb_tracker, build_name=None):
         fail_name = 'fail.{}.{}'.format(self.name, self.test.id)
         self.fail_path = pav_cfg.working_dir/'builds'/fail_name
 
+        # Don't allow syntax that may cause a file to be written outside of the
+        # build context directory.
+        files_to_create = self._config.get('create_files')
+        if files_to_create:
+            for file, contents in files_to_create.items():
+                if '../' in str(file):
+                    raise TestBuilderError("'create_file: {}': dangerous syntax"
+                                           "'..'".format(file))
```

Comment:
Checking for '../' isn't good enough. You also need to check that the path doesn't start with a '/', because 
combining '/foo' with '/bar' gets you '/bar', not '/foo/bar'.

Toss this into utils.py:
```python
def dir_contains(file, directory):
    """Check if 'file' is or is contained by 'directory'."""
    file = pathlib.Path(file).resolve()
    directory = pathlib.Path(directory).resolve()
    while file.parent != file:
        if file == directory:
            return True
        file = file.parent
    return False
```